internal/template/template_test.go

```diff
@@ -304,26 +304,28 @@ func TestSubstr(t *testing.T) {
 		input         string
 		start, length int
 		want          string
+		wantErr       bool
 	}{
-		{"sample string", -20, 0, ""},
-		{"sample string", 3, -3, ""},
-		{"sample string", 100, 4, ""},
-		{"sample string", 0, 40, "sample string"},
-		{"sample string", 0, 5, "sampl"},
-		{"sample string", 5, 4, "e st"},
-		{"sample string", 3, 0, ""},
+		{"sample string", -20, 0, "", true},
+		{"sample string", 3, -3, "", true},
+		{"sample string", 100, 4, "", true},
+		{"sample string", 0, 40, "sample string", false},
+		{"sample string", 0, 5, "sampl", false},
+		{"sample string", 5, 4, "e st", false},
+		{"sample string", 3, 0, "", false},
 	}
 	for _, tc := range tests {
 		got, err := substr(tc.input, tc.start, tc.length)
-		var tcValid = tc.start >= 0 && tc.start < len(tc.input) && tc.length >= 0
 
-		if !tcValid {
-			if err == nil {
-				t.Errorf("Expected error but received nil for input:\"%s\", "+
-					"start:%d, length:%d", tc.input, tc.start, tc.length)
-			} else {
-				continue
-			}
+		if err != nil && !tc.wantErr {
+			t.Errorf("received error for valid input:\"%s\", start:%d, "+
```

Comment:
you can use t.Fatalf instead of Errorf + continue, similar to below

Reply:
also generally its good practice to print out what the err is

Reply:
Done.